src/client.ts

```diff
@@ -5456,27 +5462,36 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
      * @return {Promise} Resolves:
      *    {@link module:models/event-timeline~EventTimeline} timeline with the latest events in the room
      */
-    public async getLatestTimeline(timelineSet: EventTimelineSet): Promise<EventTimeline> {
+    public async getLatestTimeline(timelineSet: EventTimelineSet): Promise<Optional<EventTimeline>> {
         // don't allow any timeline support unless it's been enabled.
         if (!this.timelineSupport) {
             throw new Error("timeline support is disabled. Set the 'timelineSupport'" +
                 " parameter to true when creating MatrixClient to enable it.");
         }
 
-        const messagesPath = utils.encodeUri(
-            "/rooms/$roomId/messages", {
-                $roomId: timelineSet.room.roomId,
-            },
-        );
-
-        const params: Record<string, string | string[]> = {
-            dir: 'b',
-        };
-        if (this.clientOpts.lazyLoadMembers) {
-            params.filter = JSON.stringify(Filter.LAZY_LOADING_MESSAGES_FILTER);
+        if (!timelineSet.room) {
+            throw new Error("getLatestTimeline only supports room timelines");
         }
 
-        const res = await this.http.authedRequest<IMessagesResponse>(undefined, Method.Get, messagesPath, params);
+        let res: IMessagesResponse;
+        const roomId = timelineSet.room?.roomId;
```

Comment:
```suggestion
        const roomId = timelineSet.room.roomId;
```

since the check just above this ensures that `timelineSet.room` is truthy